### PR TITLE
Remove obsolete files

### PR DESCRIPTION
**Pull request by sciascid**
* sql/wsrep_tc.h - empty file, which was committed in by mistake.
* sql/wsrep_hton.cc - wsrep handlerton is no longer used by galera 4.

### CODE REVIEW

File: sql/wsrep_hton.cc [hunk 1]
```diff
@@ -1,659 +0,0 @@
...
-extern int binlog_close_connection(handlerton *hton, THD *thd);
-extern ulonglong thd_to_trx_id(THD *thd);
-
-extern "C" int thd_binlog_format(const MYSQL_THD thd);
-// todo: share interface with ha_innodb.c
-
-/*
-  Cleanup after local transaction commit/rollback, replay or TOI.
-*/
-void wsrep_cleanup_transaction(THD *thd)
```
Comment (SachinSetiya) [inline sql/wsrep_hton.cc]: wsrep_cleanup_transaction still present in wsrep_mysqld.h

File: sql/wsrep_hton.cc [hunk 2]
```diff
@@ -1,659 +0,0 @@
...
-      }
-    }
-  }
-}
-
-/*
-  Calls wsrep->post_commit() for locally executed transactions that have
-  got seqno from provider (must commit) and don't require replaying.
- */
-void wsrep_post_commit(THD* thd, bool all)
```
Comment (SachinSetiya) [inline sql/wsrep_hton.cc]: it is called in sql_load.cc

File: sql/wsrep_hton.cc [hunk 3]
```diff
@@ -1,659 +0,0 @@
...
-  mysql_mutex_unlock(&thd->LOCK_thd_data);
-  DBUG_RETURN(0);
-}
-
-
-extern Rpl_filter* binlog_filter;
-extern my_bool opt_log_slave_updates;
-
-enum wsrep_trx_status
-wsrep_run_wsrep_commit(THD *thd, bool all)
```
Comment (SachinSetiya) [inline sql/wsrep_hton.cc]: still present in sql_load.cc

**Comment by sciascid** [comment]
@SachinSetiya Thanks for your review. I addressed your comments in a follow up commit.

About the comments in wsrep_mysqld.h: I removed most of them. But turns out that some are needed to make mariadb compile WITH_WSREP=OFF. Those I restored.

**Comment by SachinSetiya** [comment]
@sciascid , Thanks for addressing comments, It compiles with wsrep_on and off/

**Pull request merged**